### PR TITLE
Improve hint for as_ref_mut

### PR DESCRIPTION
**Pull request by Frosthage**
The hint for as_ref_mut was written before the function ```num_sq``` was added and is currently incorrect. I'm pretty sure you can improve this hint, but at least it's correct

### CODE REVIEW

File: info.toml [hunk 1]
```diff
@@ -1135,4 +1135,4 @@ name = "as_ref_mut"
 path = "exercises/conversions/as_ref_mut.rs"
 mode = "test"
 hint = """
-Add AsRef<str> as a trait bound to the functions."""
+Add AsRef<str> or AsMut<u32> as a trait bound to the functions."""
```
[the remark this answers is not in the document]
Comment (CaliViking) [reply info.toml:1138]: Why would you limit the trait to u32? Shouldn't the trait allow any parameter that can be squared, so Mul trait? I know that the unit test specifies u32, but that seems pretty narrow, shouldn't the function support i32, i64, u32, u63, f32, f64?

[the remark this answers is not in the document]
Comment (diannasoreil) [reply info.toml:1138]: It's just as an example 🤷

**Comment by Ttibsi** [comment]
Just realised this existed and went to put in my own PR instead (That I've now closed) - @Frosthage Instead of straight up adding the answer there (and also limiting it to a u32, like @CaliViking mentioned above), what if it had it's own hint saying something like "Consider the type you'd want to use" - I think it's already clear (ish) from the actual comments on that file that it wants to be `T: AsMut<_>` like the others are something like `T: AsRef<str>` and the hint just needs to steer the user towards the correct data type. I'm not sure what the data type we could hint towards should be - I used `u32` in my own solution, but any numeric one would be fine (does this pass with a generic `usize`? I've not tested that)

**Comment by njr0** [comment]
FWIW, I just went through all the exercises (as a genuine newbie to Rust) and this was the only exercise I failed to complete properly (as far as I can tell), and I found the hint quite unhelpful.

I could fix it if I specialised `num_sq`  to u32 (with or without the Box), but trying to do with with a combination of generic and Box seemed hard, especially since a lot of types don't implement multiplication. I'd love to see what a correct solution with the generic, handling the Box.

File: info.toml (hunk 1 above)
[the remark this answers is not in the document]
Comment (godmir) [reply info.toml:1138]: You can actually make it generic, e.g.:

fn num_sq<T: AsMut<U> + AsRef<U>, U>(arg: &mut T) where U: Copy + MulAssign<U> {
    let x = *arg.as_ref();
    *arg.as_mut() *= x;
}

**Comment by shadows-withal** [comment]
@all-contributors please add @Frosthage for content

**Pull request merged**